[20기 문성호] 회원가입 기능

Room9 opened the pull request:
>wecode PR message template 및 체크 리스트 입니다. 
아래 사항들을 전부 작성/체크 하시고 PR 해주세요!

## 수정 사항 간략한 한줄 요약
(여기에 수정 사항에 대한 간략한 한줄 요약/제목 작성해 주세요.)
- wesop 회원가입 기능 구현 
		 
## 수정 사항들 자세한 내용
(여기에 수정 사항에 대한 자세한 내용을 작성해 주세요.)
- 이메일 유효성검사 : @포함 / .com으로 끝나야 함
- 패스워드 유효성 검사: 대문자, 숫자 1개 이상 포함 / 6글자 이상 10글자 이하
- 필수 입력 사항: 이메일, 비밀번호, 성, 이름
- 비필수 입력사항: 전화번호
- 회원가입 등록 완료 시 자동로그인되어 jwt return

## 기타 질문 및 특이 사항
(궁금한 사항들, 하면서 느낀 점들 공유해주실 것이 있으면 자유롭게 작성해 주세요.)
- 함수 로직이 너무 긴 것 같습니다. 좀 더 효율적인 방법이 없을지 궁금합니다.(예시) 

## 체크 리스트 (아래 사항들이 전부 체크되어야만 merge가 됩니다!)
- [x] 필요한 test들을 완료하였고 기능이 제대로 실행되는지 확인 하였습니다.
- [x] Wecode의 코드 스타일 가이드에 맞추어 코드를 작성 하였습니다.
- [x] 제가 의도한 파일들과 수정 사항들만 커밋이 된 것을 확인 하였습니다.
- [x] 본 수정 사항들을 팀원들과 사전에 상의하였고 팀원들 모두 해당 PR에 대하여 알고 있습니다.

Commit:
ADD: 회원가입 구현
- 이메일, 패스워드 유효성 검사 포함
- 회원가입 완료 즉시 로그인 상태 전환 위해 jwt 리턴

Commit:
ADD: 회원가입 구현
- 이메일, 패스워드 유효성 검사 포함
- 회원가입 완료 즉시 로그인 상태 전환 위해 jwt 리턴

jamessoun93 commented on users/views.py:
```diff
@@ -1,3 +1,59 @@
+                bcrypt.gensalt()
+            ).decode('utf-8')
+
+            user = User.objects.create(
+                email       =email,
+                password    =hash_password,
+                first_name  =first_name,
+                last_name   =last_name,
+                phone = phone_number
+                )
```
`=` 기준 정렬 조금 신경써서 해주세요!
```suggestion
            user = User.objects.create(
                email      = email,
                password   = hash_password,
                first_name = first_name,
                last_name  = last_name,
                phone      = phone_number
                )
```

Room9 replied:
넵 신경쓰겠습니다

jamessoun93 commented on users/views.py:
```diff
@@ -1,3 +1,59 @@
+                phone = phone_number
+                )
+
+            access_token = jwt.encode(
+                    {'user_id' : user.id}, 
+                    SECRET, 
+                    algorithm = 'HS256'
+                )
+
+            return JsonResponse({'token':access_token}, status=201)
```
회원가입 로직에 토큰 로직이 왜 있을까요?

Room9 replied:
aesop 사이트 회원가입 완료 시 자동으로 해당 계정으로 로그인되어 jwt return 추가하였습니다

Commit:
refactor: 리뷰사항 수정
- 기준 정렬 재 확인 완료
- branch를 새로 따는 걸 깜빡하고 로그인 및 추가페이지 작업 진행하는라 추가 된 사항이 많습니다ㅠ

merge 해주시면 pull 해서 후속 진행 하겠습니다

jamessoun93 commented:
LGTM
